### PR TITLE
Fix unit test warnings

### PR DESCRIPTION
**Pull request by ihor-sviziev**
### Description (*)

1. Fixed warnings
2. Renamed all mock variables to have `Mock` suffix
3. Removed using deprecated Object Manager class
4. Changed scope from protected to private for properties and methods
5. Added return types for methods

**Note**: I don't have access to EE/B2b version, so I can't fix warnings in the `Magento\SharedCatalog` module.

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/33695

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run Static Tests, Unit Tests

File: dev/tests/unit/phpunit.xml.dist [hunk 1]
```diff
@@ -28,6 +28,7 @@
         </testsuite>
     </testsuites>
     <php>
+        <ini name="memory_limit" value="-1"/>
```
Comment (ihor-sviziev) [inline dev/tests/unit/phpunit.xml.dist:31]: I made this change to remove the memory limit. Earlier, the command `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist` was failing with fatal error:
image
After this change, the issue is gone:
image

PS: I think it uses too much memory in unit tests, but that should be investigated separately

File: app/code/Magento/Wishlist/Test/Unit/Controller/Index/UpdateItemOptionsTest.php [hunk 2]
```diff
@@ -155,93 +150,75 @@ protected function tearDown(): void
...
         $this->resultRedirectMock->expects($this->once())
             ->method('setPath')
             ->with('*/*/')
             ->willReturnSelf();
 
-        $controller = new Remove(
-            $this->context,
-            $this->wishlistProvider,
-            $this->formKeyValidator
-        );
```
Comment (ihor-sviziev) [inline app/code/Magento/Wishlist/Test/Unit/Controller/Index/UpdateItemOptionsTest.php:232]: There was a testing of an absolutely different controller, which is a mistake. That controller is already covered with such a case.

**Comment by ihor-sviziev** [comment]
@magento run Static Tests, Unit Tests

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento run all tests

just to be sure (and include SVC and MHI)

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, WebAPI Tests, Integration Tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Integration Tests

**Review by andrewbess — APPROVED** [review]

**Review by eduard13 — APPROVED** [review]
👍

**Comment by ihor-sviziev** [comment]
@magento run Integration Tests

**Pull request merged**